### DISCUSSION TITLE
Reversing incorrect CALL_RETURNs annotations across project

### DISCUSSION THREAD

**Discussion by Joey35233** [NationalSecurityAgency/ghidra#6778] [Q&A]
Hi, I'm not sure if this is (just?) an actual issue so I'll start here with it. A few years ago, we set up our project and didn't catch that a few functions were marked as CALL_RETURN every time they showed up in the disassembly, which was tens of thousands of times. This has left huge holes where the executable hasn't been analyzed. You can see the issue below. Unfortunately, running Analysis again does nothing.

![image](https://github.com/user-attachments/assets/c4197a36-9afd-4fee-b31a-637452e10f97)

Just trying to see what we can do about it. My basic thought is to search for these instances with a script, set CALL_RETURN to CALL, and then run Analysis again, but I'm hoping there's a better way.

**Comment by mumbel** [comment]
check if `Set` got marked `noreturn` by auto-analysis

**Comment by Joey35233** [comment]
Sorry, I believe they were marked as `noreturn` but I've long since manually unset that attribute and it remains stuck.
